### ISSUE TITLE
TypeError: Cannot read property 'match' of null

### ISSUE THREAD

**Issue by bkimminich** [mashpie/i18n-node#419]
```
C:\Data\GitHub\juice-shop\node_modules\i18n\i18n.js:643
    if (extension && !filename.match(extensionRegex)) return false;
                               ^

TypeError: Cannot read property 'match' of null
    at guessLocaleFromFile (C:\Data\GitHub\juice-shop\node_modules\i18n\i18n.js:643:32)
    at FSWatcher.<anonymous> (C:\Data\GitHub\juice-shop\node_modules\i18n\i18n.js:166:32)
    at FSWatcher.emit (events.js:189:13)
    at FSEvent.FSWatcher._handle.onchange (internal/fs/watchers.js:131:12)
```

This happens when a string is queried that does not exist (yet) in the given locale. When logging `filename` in the function throwing the error, I get e.g. `de_DE.json.tmp` and next up `null`, whereas otherwise the function receives `de_DE.json` as I would always expect.

My config is

```
i18n.configure({
  locales: locales.map(locale => locale.key),
  directory: path.join(__dirname, '/i18n'),
  cookie: 'language',
  defaultLocale: 'en',
  autoReload: true
})
app.use(i18n.init)
```

**Comment by bkimminich** [comment]
More precisely, this is the sequence of events when `fs.watch` triggers in my error case after adding a `console.log(event + ' -> ' + filename)` log statement:

```
rename -> de_DE.json.tmp
change -> null
```

**Comment by TyrionGraphiste** [comment]
The issue is still present on the latest version. In my case it fails when using "autoReload" to "true".

**Comment by mashpie** [comment]
@TyrionGraphiste any more details to reproduce?

**Comment by TyrionGraphiste** [comment]
Hi @mashpie , I've no clue about what is causing this issue, it seems the "filename" parameter is null in my case. Adding `if (!filename) return false;` apparently solves the issue but it might not be a real and solid workaround.

**Comment by TyrionGraphiste** [comment]
I didn't see it was fixed in the 0.8.5 version and the "Open" tag on this ticket led me to think I had the latest version. I was still on 0.8.4 version, passed to 0.8.5 and it's fixed. Sorry for this mistake!

**Comment by mashpie** [comment]
Ok, no worries glad to hear it‘s fixed.
